openpgp/mlkem_ecdh/mlkem_ecdh.go

```diff
@@ -141,34 +141,26 @@ func Decrypt(priv *PrivateKey, kEphemeral, ecEphemeral, ciphertext []byte) (msg
 
 // buildKey implements the composite KDF from
 // https://github.com/openpgp-pqc/draft-openpgp-pqc/pull/161
-func buildKey(pub *PublicKey, eccSecretPoint, eccEphemeral, eccPublicKey, mlkemKeyShare, mlkemEphemeral []byte, mlkemPublicKey kem.PublicKey) ([]byte, error) {
+func buildKey(pub *PublicKey, eccSecretPoint, eccEphemeral, eccPublicKey, mlkemKeyShare []byte) ([]byte, error) {
 	/// Set the output `ecdhKeyShare` to `eccSecretPoint`
 	eccKeyShare := eccSecretPoint
 
-	serializedMlkemPublicKey, err := mlkemPublicKey.MarshalBinary()
-	if err != nil {
-		return nil, err
-	}
-
 	//   mlkemKeyShare   - the ML-KEM key share encoded as an octet string
-	//   mlkemEphemeral  - the ML-KEM ciphertext encoded as an octet string
-	//   mlkemPublicKey  - The ML-KEM public key of the recipient as an octet string
 	//   algId           - the OpenPGP algorithm ID of the public-key encryption algorithm
 	//   eccKeyShare     - the ECDH key share encoded as an octet string
 	//   eccEphemeral    - the ECDH ciphertext encoded as an octet string
 	//   eccPublicKey    - The ECDH public key of the recipient as an octet string
 
 	// SHA3-256(mlkemKeyShare || eccKeyShare || eccEphemeral || eccPublicKey ||
-	//          mlkemEphemeral || mlkemPublicKey || algId || "OpenPGPCompositeKDFv1")
+	//          algId || "OpenPGPCompositeKDFv1" || len("OpenPGPCompositeKDFv1"))
 	h := sha3.New256()
 	_, _ = h.Write(mlkemKeyShare)
 	_, _ = h.Write(eccKeyShare)
 	_, _ = h.Write(eccEphemeral)
 	_, _ = h.Write(eccPublicKey)
-	_, _ = h.Write(mlkemEphemeral)
-	_, _ = h.Write(serializedMlkemPublicKey)
 	_, _ = h.Write([]byte{pub.AlgId})
 	_, _ = h.Write([]byte(kdfContext))
+	_, _ = h.Write([]byte{21})
```

Comment:
```suggestion
	_, _ = h.Write([]byte{len(kdfContext)})
```
The `len` of a string constant is also a constant so this shouldn't hurt performance